### PR TITLE
Abidlabs/tests analytics fix

### PR DESCRIPTION
**Pull request by abidlabs**
Fixed tests so they do not trigger our analytics

### CODE REVIEW

File: gradio/interface.py [hunk 1]
```diff
@@ -171,12 +171,20 @@ def __init__(self, fn, inputs=None, outputs=None, verbose=False, examples=None,
         self.server_port = server_port
         self.simple_server = None
         self.allow_screenshot = allow_screenshot
-        self.allow_flagging = os.getenv("GRADIO_FLAGGING") or allow_flagging
+        # If parameter is provided, use that; otherwise, environmet variable; otherwise, True
```
Comment (aliabid94) [inline gradio/interface.py:174]: Remove comment, self evident through code. Also, shouldn't allow_flagging be default set to None?

File: gradio/interface.py [hunk 2]
```diff
@@ -171,12 +171,20 @@ def __init__(self, fn, inputs=None, outputs=None, verbose=False, examples=None,
...
+        if allow_flagging is None:
+            self.allow_flagging = bool(os.getenv("GRADIO_FLAGGING") or True)
```
Comment (aliabid94) [inline gradio/interface.py:176]: Want to use the same format of os env logic used for theme? 
`self.theme = theme if theme is not None else os.getenv("GRADIO_THEME", "default")`
so
`self.allow_flagging = allow_flagging if allow_flagging is not None else bool(os.getenv("GRADIO_FLAGGING", "1"))`

File: gradio/interface.py [hunk 3]
```diff
@@ -171,12 +171,20 @@ def __init__(self, fn, inputs=None, outputs=None, verbose=False, examples=None,
...
+        else:
+            self.allow_flagging = allow_flagging
         self.flagging_options = flagging_options
         self.flagging_dir = flagging_dir
         self.encrypt = encrypt
         Interface.instances.add(self)
-        self.analytics_enabled = analytics_enabled
+        # If parameter is provided, use that; otherwise, environmet variable; otherwise, True
```
Comment (aliabid94) [inline gradio/interface.py:183]: Same as above

File: test/test_external.py [hunk 4]
```diff
@@ -146,25 +150,25 @@ def test_english_to_spanish(self):
 
     def test_sentiment_model(self):
         interface_info = gr.external.load_interface("models/distilbert-base-uncased-finetuned-sst-2-english", alias="sentiment_classifier")
-        io = gr.Interface(**interface_info)
+        io = gr.Interface(**interface_info, analytics_enabled=False)
```
Comment (aliabid94) [inline test/test_external.py:153]: Why do you have to do this on top of the env variable?

File: test/test_inputs.py [hunk 5]
```diff
@@ -8,7 +8,9 @@
 import os
 import tempfile
 import json
-import shutil
+
+
+os.environ["GRADIO_ANALYTICS_ENABLED"] = ""  # Disables analytics
```
Comment (aliabid94) [inline test/test_inputs.py:13]: You overuse comments. If it's unclear that this line disables analytics (which I agree is unclear), then do something like `os.environ["GRADIO_ANALYTICS_ENABLED"] = "False"`

Comment (abidlabs) [reply test/test_inputs.py:13]: That actually doesn't work. `bool("False") = True` so the standard way to set an env variable as `False` is by setting it to `""`

File: gradio/interface.py (hunk 2 above)
Comment (abidlabs) [reply gradio/interface.py:176]: Sure

File: gradio/interface.py (hunk 1 above)
Comment (abidlabs) [reply gradio/interface.py:174]: Sure

File: test/test_external.py (hunk 4 above)
Comment (abidlabs) [reply test/test_external.py:153]: You don't need, I started doing this and then realized we should make it an env. I'll fix

**Pull request merged**